webui/react/src/utils/trial.ts

```diff
@@ -43,6 +45,45 @@ export const metricNameToValue = (metricName: MetricName): string => {
   return `${metricName.type}|${metricName.name}`;
 };
 
+export const metricNameFromValue = (metricValue: string): MetricName | undefined => {
+  const trainingPrefix = `${MetricType.Training}|`;
+  const validationPrefix = `${MetricType.Validation}|`;
+  if (metricValue.startsWith(trainingPrefix)) {
+    return {
+      name: metricValue.slice(trainingPrefix.length),
+      type: MetricType.Training,
+    };
+  } else if (metricValue.startsWith(validationPrefix)) {
+    return {
+      name: metricValue.slice(validationPrefix.length),
+      type: MetricType.Validation,
+    };
+  } else {
+    const errName = 'metricNameFromValueUnrecognizedMetricType';
+    const errSlug = 'metricnamefromvalue-unrecognized-metric-type';
+    const errMessage = `metricNameFromValue was called, but the metricName doesn't appear to " +
+        'be a training metric or a validation metric (${metricValue})`;
+    let silent = true;
+    if (process.env.IS_DEV) {
+      silent = false;
+    }
+    const daErr: DaError = {
+      error: {
+        message: errMessage,
+        name: errName,
+      },
+      id: errSlug,
+      isUserTriggered: false,
+      level: ErrorLevel.Error,
+      message: errMessage,
+      silent: silent,
```

Comment:
**nit:** you can shorten the silent definition by doing something like this:
```
  silent: !process.env.IS_DEV,
```

Reply:
fixed